harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -362,6 +363,28 @@ def remove_files(self, globs: List[str]) -> None:
         else:
             logging.info(f"Partial deletion of checkpoint {self.uuid} is in progress.")
 
+    def get_metrics(self, group: str = "") -> Iterable["metrics.TrialMetrics"]:
+        """
+        Gets all metrics for a given metric group associated with this checkpoint.
+        The checkpoint can be originally associated by calling
+        ``core_context.utils.report_task_using_checkpoint(<CHECKPOINT>)``
+        from within a task.
+
+        Arguments:
+            group (str, optional): Group name for the metrics (example: "training", "validation").
+                All metrics will be returned when querying by "".
```

Comment:
can we have default be `None` for all metrics?

Reply:
Agree with Ilia, but separately, do we _have_ to make it easier to get metrics for a checkpoint?

That particular coupling in our system is one I'd really like to get rid of.  Not sure how critical it is to the this particular effort though.

Reply:
> That particular coupling in our system is one I'd really like to get rid of. Not sure how critical it is to the this particular effort though.

The issue at least from my perspective is that this coupling __is__ the feature. The goal being that we wanted to allow for users to know what the inference metrics for a checkpoint/model version was over many potential runs. From a strict DB perspective, it isn't coupled as it only works through trials as a proxy, but I understand there's still some degree of functionality coupling going on here

Reply:
added this change